modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java

```diff
@@ -232,7 +232,7 @@ private void logForRetry(Level level, String action, Exception e) {
                 start + currentOffset,
                 purpose.getKey(),
                 attempt,
-                currentStreamProgress,
+                Math.subtractExact(Math.addExact(start, currentOffset), currentStreamFirstOffset),
```

Comment:
I'd be inclined to extract this as a private method rather than duplicate the expression, especially since inlining it like this loses the name that explains what it really means.

Reply:
Makes sense. See 9d31e189677ffe358fcc3e69471cff3c4d48a741